core/modules/system/system.install

```diff
@@ -98,6 +98,18 @@ function system_requirements($phase) {
     $requirements['php_register_globals']['value'] = $t('Disabled');
   }
 
+  // Test PHP max_input_vars setting.
+  $max_input_vars = ini_get('max_input_vars');
+  $requirements['php_max_input_vars'] = array(
+    'title' => $t('PHP form value limit'),
+    'value' => $t("PHP limits the number of values in a single form to @value items.", array('@value' => number_format($max_input_vars, 0, '.', ','))),
+  );
+  // 1000 is the default value, and indicates potential issues.
+  if ($max_input_vars <= 1000) {
+    $requirements['php_max_input_vars']['description'] = $t('When set too low, PHP\'s <code>max_input_vars</code> setting can cause silent failures on large forms. Use caution when using the permissions page, ordering large menus, arranging large vocabularies, or editing content with lots of fields.');
+    $requirements['php_max_input_vars']['severity'] = REQUIREMENT_ERROR;
```

Comment:
Given that not all sites are affected by this, should we change this to a warning instead? ...or perhaps leave it be an error if bellow 1000, and a warning if between 1001 and 10007? We could have the message say that a safe value is anything above 10k.